spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -131,6 +131,7 @@ public void doWith(Method method) throws IllegalArgumentException {
 						inputHolders.put(name, new ChannelHolder(createBindableChannel(name, channelType), true));
 					}
 					else {
+						inputHolders.put(name, new ChannelHolder(sharedChannel, false));
```

Comment:
Since the channel holders are only used when binding inputs/outputs for the Bindable non-shared channels, I believe we need not have to add the `sharedChannel` into the input/output ChannelHolders.